lib/spack/spack/parser.py

```diff
@@ -80,8 +82,13 @@
 
 HASH = r"[a-zA-Z_0-9]+"
 
-#: A filename starts either with a "." or a "/" or a "{name}/"
-FILENAME = r"(\.|\/|[a-zA-Z0-9-_]*\/)([a-zA-Z0-9-_\.\/]*)(\.json|\.yaml)"
+#: A filename starts either with a "." or a "/" or a "{name}/,
+# or on Windows, a drive letter followed by a colon and "\"
+# or "." or {name}\
+if not IS_WINDOWS:
+    FILENAME = r"(\.|\/|[a-zA-Z0-9-_]*\/)([a-zA-Z0-9-_\.\/]*)(\.json|\.yaml)"
+else:
+    FILENAME = r"(\.|[a-zA-Z0-9-_]*\\|[a-zA-Z]:\\)([a-zA-Z0-9-_\.\\]*)(\.json|\.yaml)"
```

Comment:
Sounds good I'll set that up. Thanks for the offer, I'll reach out if anything unexpected happens.

Reply:
This test has been setup so that it runs essentially the same test on unix/windows and skips the platform it's not on at test time. There's a good amount of code duplication caused by this setup, so this is fine for now, but I'm working on setting up some class or utility that will allow us to create cross platform tests with less code duplication. Pytest offers some pretty need features when working with pytest meta-classes so I think that will help similar cases going forward.